lib/ansible/plugins/connection/__init__.py

```diff
@@ -254,10 +254,8 @@ def check_password_prompt(self, b_output):
             return False
         elif isinstance(self._play_context.prompt, string_types):
             b_prompt = to_bytes(self._play_context.prompt).strip()
-            b_lines = b_output.splitlines(True)
-            if not b_lines:
-                return False
-            return b_lines[-1].strip().endswith(b_prompt) or b_lines[0].strip().endswith(b_prompt)
+            b_lines = b_output.splitlines()
+            return any(l.strip().startswith(b_prompt) for l in b_lines)
```

Comment:
tempted to 'reverse' the order as the prompt is more likely to exist in the last portion of the output